taskvine/src/manager/vine_schedule.c

```diff
@@ -149,6 +149,46 @@ int check_worker_have_enough_disk_with_inputs(struct vine_manager *q, struct vin
 	return ok;
 }
 
+/* Check if any of the resources is defined and committable on a given worker.
+ * @param q         Manager info structure
+ * @param resource  @vine_resources.h:struct vine_resources
+ * @return 1 if the resource type is defined and can be allocated to a task, 0 otherwise.
+ */
+static int is_resource_committable(struct vine_manager *q, struct vine_resource resource)
```

Comment:
You are unnecessarily copying the `struct vine_resource` data. In this case, I think it will be clearer if you do the ` resource.total > 0 && resource.inuse < overcommitted_resource_total(q, resource.total);` per resource inside `check_worker_have_committable_resources`.